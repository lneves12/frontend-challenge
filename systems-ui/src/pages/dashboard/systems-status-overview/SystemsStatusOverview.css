.container {
  height: 450px;
  display: flex;
  justify-content: flex-start;
}

.filtersGroup {
  margin-top: 50px;
}
