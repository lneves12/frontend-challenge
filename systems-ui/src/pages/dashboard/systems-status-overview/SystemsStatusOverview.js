import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {VictoryPie} from "victory";
import styles from "./SystemsStatusOverview.css";
import classnames from "classnames";

class SystemsStatusOverview extends Component {

  render() {
    let pieOverview = [
      {id: 1, x: "Free", y: 0},
      {id: 2, x: "Busy", y: 0},
      {id: 3, x: "Down", y: 0},
    ]

    this.props.systems.forEach((system) => {
      const group = pieOverview.find((group) => group.id === +system.provision_state_id);
      if(group) {
        group.y += 1;
      }
    });

    return (
      <div className="row">
        <div className={classnames("col-3", styles.filtersGroup)}>
          <legend>System State</legend>
          <div className="form-check form-check-inline">
            <input className="form-check-input"
                   type="radio" name="inlineRadioOptions"
                   id="inlineRadio1"
                   value="last_boot"
                   onClick={() => this.props.onFilterBy("provision_state_id", 1)} />
            <label className="form-check-label" htmlFor="inlineRadio1">Free</label>
          </div>
          <div className="form-check form-check-inline">
            <input className="form-check-input"
                   type="radio"
                   name="inlineRadioOptions"
                   id="inlineRadio2"
                   value="modified"
                   onClick={() => this.props.onFilterBy("provision_state_id", 2)} />
            <label className="form-check-label" htmlFor="inlineRadio2">Busy</label>
          </div>
          <div className="form-check form-check-inline">
            <input className="form-check-input"
                   type="radio"
                   name="inlineRadioOptions"
                   id="inlineRadio2"
                   value="modified"
                   onClick={() => this.props.onFilterBy("provision_state_id", 3)} />
            <label className="form-check-label" htmlFor="inlineRadio2">Down</label>
          </div>
        </div>
        <div className={classnames("col-6", styles.container)}>
          <VictoryPie
            animate={{ duration: 1000 }}
            colorScale={["#28a745", "#ffc107", "#dc3545" ]}
            data={pieOverview} />
        </div>
        <div className={classnames("col-2", styles.filtersGroup)}>
          <dt>Last Updated</dt>
          <dd>{this.props.lastUpdated}</dd>
        </div>
      </div>
    );
  }
}

SystemsStatusOverview.propTypes = {
  systems: PropTypes.array.isRequired,
  lastUpdated: PropTypes.string.isRequired,
  onFilterBy: PropTypes.func.isRequired,
};

export default SystemsStatusOverview;
