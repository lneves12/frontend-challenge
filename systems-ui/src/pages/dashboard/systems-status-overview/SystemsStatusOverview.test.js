import React from 'react';
import renderer from 'react-test-renderer';
import SystemsStatusOverview from "./SystemsStatusOverview";

const mockSystems = [{
  "id": "1000010000",
  "org_id": 1,
  "digital_server_id": "ID-1000010000",
  "server_arch_id": 1029,
  "os": "sles-release",
  "release": "12.3",
  "name": "suma-ref31-cli-sles12sp3-1",
  "description": "Initial Registration Parameters:nOS: sles-releasenRelease: 12.3nCPU Arch: x86_64",
  "info": "",
  "secret": "d910e6d5b694dd01cdb5e9a042b92a20bc09aae85342182b971cf0cc107143a1",
  "creator_id": 1,
  "auto_update": "N",
  "contact_method_id": 0,
  "running_kernel": "4.4.73-5-default",
  "last_boot": 1516800657040,
  "provision_state_id": 2,
  "channels_changed": 1516800657040,
  "cobbler_id": "",
  "machine_id": "a2a7915642f3efeec4837a015a679bc1",
  "hostname": "suma-ref31-cli-sles12sp3-1",
  "created": 1516800657040,
  "modified": 1516800657040
},
  {
    "id": "1000010001",
    "org_id": 1,
    "digital_server_id": "ID-1000010001",
    "server_arch_id": 1029,
    "os": "sles-release",
    "release": "12.3",
    "name": "suma-ref31-cli-sles12sp3-2",
    "description": "Initial Registration Parameters:nOS: sles-releasenRelease: 12.3nCPU Arch: x86_64",
    "info": "",
    "secret": "d910e6d5b694dd01cdb5e9a042b92a20bc09aae85342182b971cf0cc107143a2",
    "creator_id": 1,
    "auto_update": "N",
    "contact_method_id": 0,
    "running_kernel": "4.4.73-5-default",
    "last_boot": 1516800657040,
    "provision_state_id": 1,
    "channels_changed": 1516800657040,
    "cobbler_id": "",
    "machine_id": "a2a7915642f3efeec4837a015a679bc2",
    "hostname": "suma-ref31-cli-sles12sp3-2",
    "created": 1516800657040,
    "modified": 1516800657040
  },
];

describe('renders correctly', () => {
  it('renders correctly with filters ', () => {
    const tree = renderer.create(
      <SystemsStatusOverview
        systems={mockSystems}
        lastUpdated={'1516800657040'}
        onFilterBy={jest.fn()}
        selectedFilters={{
          provision_state_id: [1, 2]
        }}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without filters ', () => {
    const tree = renderer.create(
      <SystemsStatusOverview
        systems={mockSystems}
        lastUpdated={'1516800657040'}
        onFilterBy={jest.fn()}
        selectedFilters={{}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
