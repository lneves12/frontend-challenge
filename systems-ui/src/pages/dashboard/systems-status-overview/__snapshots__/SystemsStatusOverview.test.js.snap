// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders correctly with filters  1`] = `
<div
  className="row"
>
  <div
    className="col-3"
  >
    <legend>
      System State
    </legend>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        defaultChecked={true}
        id="inlineCheckbox1"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="inlineCheckbox1"
      >
        Free
      </label>
    </div>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        defaultChecked={true}
        id="inlineCheckbox2"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="inlineCheckbox2"
      >
        Busy
      </label>
    </div>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        defaultChecked={false}
        id="inlineCheckbox3"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="inlineCheckbox3"
      >
        Down
      </label>
    </div>
  </div>
  <div
    className="col-6"
  >
    <div
      className="VictoryContainer"
      style={
        Object {
          "height": "100%",
          "pointerEvents": "none",
          "position": "relative",
          "touchAction": "none",
          "width": "100%",
        }
      }
    >
      <svg
        aria-labelledby="victory-container-1-title victory-container-1-desc"
        height={400}
        role="img"
        style={
          Object {
            "height": "100%",
            "pointerEvents": "all",
            "width": "100%",
          }
        }
        viewBox="0 0 400 400"
        width={400}
      >
        <g>
          <path
            className={undefined}
            d="M9.184850993605149e-15,-150A150,150,0,1,1,9.184850993605149e-15,150L0,0Z"
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "eventKey": 0,
                "fill": "#28a745",
                "id": 1,
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
                "xName": "Free",
              }
            }
            transform="translate(200, 200)"
          />
          <path
            className={undefined}
            d="M9.184850993605149e-15,150A150,150,0,1,1,-2.7554552980815446e-14,-150L0,0Z"
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "eventKey": 1,
                "fill": "#ffc107",
                "id": 2,
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
                "xName": "Busy",
              }
            }
            transform="translate(200, 200)"
          />
          <path
            className={undefined}
            d="M-2.7554552980815446e-14,-150L0,0Z"
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "eventKey": 2,
                "fill": "#dc3545",
                "id": 3,
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
                "xName": "Down",
              }
            }
            transform="translate(200, 200)"
          />
          <text
            className={undefined}
            dx={0}
            dy={4.97}
            transform={undefined}
            x={370}
            y={200}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#252525",
                  "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                  "fontSize": 14,
                  "letterSpacing": "normal",
                  "padding": 20,
                  "stroke": "transparent",
                }
              }
              textAnchor="start"
              x={370}
            >
              Free
            </tspan>
          </text>
          <text
            className={undefined}
            dx={0}
            dy={4.97}
            transform={undefined}
            x={30}
            y={200}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#252525",
                  "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                  "fontSize": 14,
                  "letterSpacing": "normal",
                  "padding": 20,
                  "stroke": "transparent",
                }
              }
              textAnchor="end"
              x={30}
            >
              Busy
            </tspan>
          </text>
          <text
            className={undefined}
            dx={0}
            dy={-2.0300000000000002}
            transform={undefined}
            x={200}
            y={30}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#252525",
                  "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                  "fontSize": 14,
                  "letterSpacing": "normal",
                  "padding": 20,
                  "stroke": "transparent",
                }
              }
              textAnchor="middle"
              x={200}
            >
              Down
            </tspan>
          </text>
        </g>
      </svg>
      <div
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
            "zIndex": 99,
          }
        }
      >
        <svg
          height={400}
          style={
            Object {
              "height": "100%",
              "overflow": "visible",
              "width": "100%",
            }
          }
          viewBox="0 0 400 400"
          width={400}
        />
      </div>
    </div>
  </div>
  <div
    className="col-2"
  >
    <dt>
      Last Updated
    </dt>
    <dd>
      1516800657040
    </dd>
  </div>
</div>
`;

exports[`renders correctly renders correctly without filters  1`] = `
<div
  className="row"
>
  <div
    className="col-3"
  >
    <legend>
      System State
    </legend>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        defaultChecked={undefined}
        id="inlineCheckbox1"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="inlineCheckbox1"
      >
        Free
      </label>
    </div>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        defaultChecked={undefined}
        id="inlineCheckbox2"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="inlineCheckbox2"
      >
        Busy
      </label>
    </div>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        defaultChecked={undefined}
        id="inlineCheckbox3"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="inlineCheckbox3"
      >
        Down
      </label>
    </div>
  </div>
  <div
    className="col-6"
  >
    <div
      className="VictoryContainer"
      style={
        Object {
          "height": "100%",
          "pointerEvents": "none",
          "position": "relative",
          "touchAction": "none",
          "width": "100%",
        }
      }
    >
      <svg
        aria-labelledby="victory-container-2-title victory-container-2-desc"
        height={400}
        role="img"
        style={
          Object {
            "height": "100%",
            "pointerEvents": "all",
            "width": "100%",
          }
        }
        viewBox="0 0 400 400"
        width={400}
      >
        <g>
          <path
            className={undefined}
            d="M9.184850993605149e-15,-150A150,150,0,1,1,9.184850993605149e-15,150L0,0Z"
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "eventKey": 0,
                "fill": "#28a745",
                "id": 1,
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
                "xName": "Free",
              }
            }
            transform="translate(200, 200)"
          />
          <path
            className={undefined}
            d="M9.184850993605149e-15,150A150,150,0,1,1,-2.7554552980815446e-14,-150L0,0Z"
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "eventKey": 1,
                "fill": "#ffc107",
                "id": 2,
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
                "xName": "Busy",
              }
            }
            transform="translate(200, 200)"
          />
          <path
            className={undefined}
            d="M-2.7554552980815446e-14,-150L0,0Z"
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "eventKey": 2,
                "fill": "#dc3545",
                "id": 3,
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
                "xName": "Down",
              }
            }
            transform="translate(200, 200)"
          />
          <text
            className={undefined}
            dx={0}
            dy={4.97}
            transform={undefined}
            x={370}
            y={200}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#252525",
                  "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                  "fontSize": 14,
                  "letterSpacing": "normal",
                  "padding": 20,
                  "stroke": "transparent",
                }
              }
              textAnchor="start"
              x={370}
            >
              Free
            </tspan>
          </text>
          <text
            className={undefined}
            dx={0}
            dy={4.97}
            transform={undefined}
            x={30}
            y={200}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#252525",
                  "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                  "fontSize": 14,
                  "letterSpacing": "normal",
                  "padding": 20,
                  "stroke": "transparent",
                }
              }
              textAnchor="end"
              x={30}
            >
              Busy
            </tspan>
          </text>
          <text
            className={undefined}
            dx={0}
            dy={-2.0300000000000002}
            transform={undefined}
            x={200}
            y={30}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#252525",
                  "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                  "fontSize": 14,
                  "letterSpacing": "normal",
                  "padding": 20,
                  "stroke": "transparent",
                }
              }
              textAnchor="middle"
              x={200}
            >
              Down
            </tspan>
          </text>
        </g>
      </svg>
      <div
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
            "zIndex": 99,
          }
        }
      >
        <svg
          height={400}
          style={
            Object {
              "height": "100%",
              "overflow": "visible",
              "width": "100%",
            }
          }
          viewBox="0 0 400 400"
          width={400}
        />
      </div>
    </div>
  </div>
  <div
    className="col-2"
  >
    <dt>
      Last Updated
    </dt>
    <dd>
      1516800657040
    </dd>
  </div>
</div>
`;
